roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -53,6 +80,18 @@ class RooFuncWrapper final : public RooAbsReal {
          }
          _params.add(*param);
       }
+
+      // Build a wrapper over the derivative to hide clad specific types such as 'array_ref'.
+      // disable clang-format for making the following code unreadable.
+      // clang-format off
+      std::string nParams = std::to_string(_params.size());
+      std::string dWrapper = "void " + wrapperName + "(double* in, double* out) {\n"
+                             "  clad::array_ref<double> cladOut(out, " + nParams + ");\n"
+                             "  " + gradName + "(in, cladOut);\n"
+                             "}";
+      // clang-format on
+      gInterpreter->Declare(dWrapper.c_str());
+      _grad = (Grad)gInterpreter->ProcessLine((wrapperName + ";").c_str());
```

Comment:
```suggestion
      _grad = reinterpret_cast<Grad>(gInterpreter->ProcessLine((wrapperName + ";").c_str()));
```
No C-style casts please (if we have to do `reinterpret_cast` so be it) also in other places in this file.

Reply:
I think we can get away with using a static cast here?

Reply:
No, I tried that before in an other case. You can't static_cast whatever that function returns to a function pointer :(

Anyway, `reinterpret_cast` is still better than C-style cast because at least it's clear that this is a dubious cast